EIPS/eip-FOCIL.md

```diff
@@ -40,12 +40,55 @@ TBD
 - **`Slot N+1`, `t=4s`**: The attesters accept `block B` only if it includes all transactions from the local inclusion lists, or if any missing transactions cannot be appended to the end of the execution payload, or if the block is full.
 
+#### Role and participants
+
+##### IL Committee Members
+
+- **`Slot N`, `t=0 to 8s`**:
+IL committee members construct their local ILs and broadcast them over the P2P network after processing the block for slot `N` and confirming it as the head. If no block is received by `t=7s`, they should run `get_head` and build and release their local ILs based on their node’s canonical head.
+
+By default, local ILs are built by selecting raw transactions from the public mempool, ordered by priority fees, up to the local IL’s maximum size in bits (e.g., 8 kb per local IL). Additional rules can be optionally applied to maximize CR, such as prioritizing valid transactions that have been pending in the mempool the longest.
+
+##### Nodes
+
+- **`Slot N`, `t=0 to 8s`**:
+Nodes receive local ILs from the P2P network and only forward and cache those that pass the CL P2P validation rules.
+
+- **`Slot N`, `t=9s`**:, IL freeze deadline:
+Nodes freeze their local ILs view, stop forwarding and caching new local ILs.
+
+---
+
+# CL P2P Validation Rules:
+
+1. The number of transactions in the local IL does not exceed the maximum gas limit allowed.
+2. The slot of the local IL matches the current slot. Local ILs not matching the current slot should be ignored.
+3. The parent hash of the IL is recognized.
+4. The IL is received before the local IL freeze deadline (e.g., 9s) into the slot.
+5. Received two or fewer local ILs from this IL committee member (see Local IL equivocation section below).
+6. The local IL is correctly signed by the validator.
+7. The validator is part of the IL committee.
+
+---
+
+##### Proposer
+
+- **`Slot N`, `t=11s`**:
+The proposer freezes its view of local ILs and asks the EL to update its execution payload by adding transactions from its view (the exact timings will be defined after running some tests/benchmarks). Optionally, an RPC endpoint can be added to allow the proposer to request the missing local ILs from its peers (e.g., by committee index).
+
+- **`Slot N+1`, `t=0s`**:
+The proposer broadcasts its block with the up-to-date execution payload satisfying IL transactions over the P2P network.
+
+##### Attesters
+- **`Slot N+1`, `t=0 to 4s`**:
+Attesters monitor the P2P network for the proposer’s block. Upon detecting the block, they check whether all transactions from their cached local ILs are included in the proposer’s execution payload. The `Valid` function verifies if the execution payload satisfies IL validity conditions either when all transactions are present or when any missing transactions are found to be invalid when appended to the end of the payload. In these cases, attesters use the EL to conduct `nonce` and `balance` checks and verify the validity of missing transactions.
```

Comment:
It may be worth mentioning that `Valid` function is based on the frozen view of `t=9s` in the previous slot